### PR TITLE
API1800 support for Enclosure and EG

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
API1800 support for Enclosure and EG

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -50,29 +50,29 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/enclosure-groups/{id}</sub>                                                   | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/enclosure-groups/{id}</sub>                                                   | PUT         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/enclosure-groups/{id}</sub>                                                   | DELETE      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/enclosure-groups/{id}/script</sub>                                            | GET         | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   |
```
Comment (chebroluharika) [inline endpoints-support.md:58]: Please revert this as these endpoints are present in C7000

File: endpoints-support.md [hunk 2]
```diff
@@ -50,29 +50,29 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/enclosure-groups/{id}/script</sub>                                            | PUT         | :white_check_mark:   | :white_check_mark:   |  :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   |
```
Comment (chebroluharika) [inline endpoints-support.md:59]: same as above

File: endpoints-support.md [hunk 3]
```diff
@@ -50,29 +50,29 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/enclosures/{id}/environmentalConfiguration</sub>                              | PUT         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/enclosures/{id}/refreshState</sub>                                            | PUT         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/enclosures/{id}/script</sub>                                                  | GET         | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:   | :heavy_minus_sign:   |
+|<sub>/rest/enclosures/{id}/sso</sub>                                                     | GET         | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   |
```
Comment (chebroluharika) [inline endpoints-support.md:71]: same as above

File: spec/unit/resource/api1800/c7000/enclosure_spec.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::API1600::C7000::Enclosure do
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1200::C7000::Enclosure' do
+    expect(described_class).to be < OneviewSDK::API1200::C7000::Enclosure
```
Comment (chebroluharika) [inline spec/unit/resource/api1800/c7000/enclosure_spec.rb:7]: It should be API1600::C7000

File: spec/unit/resource/api1800/synergy/enclosure_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::API1600::Synergy::Enclosure do
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1200::Synergy::Enclosure' do
+    expect(described_class).to be < OneviewSDK::API1200::Synergy::Enclosure
```
Comment (chebroluharika) [inline spec/unit/resource/api1800/synergy/enclosure_spec.rb:7]: It should be API1600::Synergy

File: endpoints-support.md (hunk 1 above)
Comment (yuvirani) [reply endpoints-support.md]: Done

File: endpoints-support.md (hunk 2 above)
Comment (yuvirani) [reply endpoints-support.md]: Done

File: endpoints-support.md (hunk 3 above)
Comment (yuvirani) [reply endpoints-support.md]: Done

File: spec/unit/resource/api1800/c7000/enclosure_spec.rb (hunk 4 above)
Comment (yuvirani) [reply spec/unit/resource/api1800/c7000/enclosure_spec.rb]: Done

**Pull request merged**